### PR TITLE
modify gfw project structure #567

### PR DESCRIPTION
**Pull request by bttanakagn**
Please review #567.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -137,4 +93,72 @@
...
+    <profiles>
+        <profile>
+            <id>default</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <modules>
+                <module>terasoluna-gfw-common-libraries</module>
+                <module>terasoluna-gfw-dependencies</module>
+                <module>terasoluna-gfw-parent</module>
```
Comment (btyoshikawaa) [inline pom.xml]: - [ ] Why is `terasoluna-gfw-parent` last module?

File: .travis.yml [hunk 2]
```diff
@@ -12,5 +12,5 @@ cache:
 install: true
 
 script:
-  - ./mvn-build-all.sh clean install jacoco:report
-  - mvn -U coveralls:report
+  - mvn -U clean install jacoco:report
```
Comment (btyoshikawaa) [inline .travis.yml]: - [ ] Do you need to apply the jacoco to all of the modules yet?

For example, I would configure as follows.
* terasoluna-gfw-parent/pom.xml

    remove configuration of `maven-surefire-plugin` and `jacoco-maven-plugin`
* terasoluna-gfw-common-libraries/pom.xml

    add configuration of `maven-surefire-plugin` and `jacoco-maven-plugin`
* root pom.xml

    add configuration of `maven-surefire-plugin` and `jacoco-maven-plugin` in `common-libs-only` profile
* travis.yml
```
mvn -U clean install
mvn -U -P common-libs-only jacoco:report coveralls:report
```

File: pom.xml (hunk 1 above)
Comment (bttanakagn) [inline pom.xml]: @btyoshikawaa 
I corrected it. 4d4c2ef631283e7306c91269b7c4eb21e79a5566

Comment (btyoshikawaa) [inline pom.xml]: @bttanakagn 
Thanks.

File: .travis.yml (hunk 2 above)
Comment (bttanakagn) [inline .travis.yml]: @btyoshikawaa 
No, I don't need it. I modified these files about jacoco. 01449544e5acc1855ce3f554c1dd5640e6464d84

Comment (btyoshikawaa) [inline .travis.yml]: @bttanakagn 
I feel Good.

File: pom.xml [hunk 3]
```diff
@@ -64,70 +60,6 @@
             <url>http://repo.terasoluna.org/nexus/content/repositories/terasoluna-gfw-snapshots/</url>
         </snapshotRepository>
     </distributionManagement>
```
Comment (btyoshikawaa) [inline pom.xml]: - [ ] `distributionManagement` of root pom.xml seems unnecessary.

File: pom.xml [hunk 4]
```diff
@@ -64,70 +60,6 @@
...
-                            <sourceDirectory>terasoluna-gfw-common-libraries/terasoluna-gfw-security-web/src/main/java</sourceDirectory>
-                            <sourceDirectory>terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/main/java</sourceDirectory>
-                            <sourceDirectory>terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/main/java</sourceDirectory>
-                            <sourceDirectory>terasoluna-gfw-common-libraries/terasoluna-gfw-string/src/main/java</sourceDirectory>
-                        </sourceDirectories>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
```
Comment (btyoshikawaa) [inline pom.xml]: - [ ] Is `maven-deploy-plugin#deployAtEnd` is unnecessary?

I am concerned if example it fails to build the `terasoluna-gfw-common-libraries`, `terasoluna-gfw-parent` would have been deployed.
http://maven.apache.org/plugins/maven-deploy-plugin/deploy-mojo.html

Comment (bttanakagn) [inline pom.xml]: I verified `deployAtEnd`. It was not effective.
I put `deployAtEnd` on `terasoluna-gfw` and I insert failure test case on purpose, then some artifacts were deployed in spite of occuring error.
And I put it on `terasoluna-gfw-common-libraries`, then all common-libraries were not deployed but `terasoluna-gfw-parent` was deployed.

Comment (btyoshikawaa) [inline pom.xml]: @bttanakagn 
Thanks for your comment.
I was set to complete this tree, and will create another new issue.

**Pull request merged**